### PR TITLE
Fix @wdio/testingbot-service not using the Tunnel because of wrong import type

### PR DESCRIPTION
**Pull request by hieuxlu**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes
- Fix #6105

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)
- [X] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-testingbot-service/src/launcher.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import { performance, PerformanceObserver } from 'perf_hooks'
 import { promisify } from 'util'
 
-import testingbotTunnel from 'testingbot-tunnel-launcher'
+import testingbotTunnel = require('testingbot-tunnel-launcher')
```
Comment (christian-bromann) [inline packages/wdio-testingbot-service/src/launcher.ts:4]: Why should we switch to commonjs import just for this package?

**Review by christian-bromann — CHANGES_REQUESTED** [review]
Let's have ESM imports for TypeScript files.

File: packages/wdio-testingbot-service/src/launcher.ts (hunk 1 above)
Comment (hieuxlu) [reply packages/wdio-testingbot-service/src/launcher.ts:4]: As specified in the docs, it's the only way to work with `export =`
https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

**Comment by hieuxlu** [comment]
> Let's have ESM imports for TypeScript files.

It's the only way. Other types of import will be transpiled as `import default`, which caused the original bug because the library used `export =`
https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

File: packages/wdio-testingbot-service/src/launcher.ts (hunk 1 above)
Comment (christian-bromann) [reply packages/wdio-testingbot-service/src/launcher.ts:4]: Why do we make the change in `testingbot-tunnel-launcher.d.ts` then as well as in the mock? The [original package](https://github.com/testingbot/testingbot-tunnel-launcher/blob/master/index.js) has a normal commonjs export:

```js
module.exports = require('./build/tunnel-launcher')
```

which can be imported using

```js
import TunnelLauncher from 'testingbot-tunnel-launcher'
```

So we should try to mimic this in our TS files and mocks.

**Review by christian-bromann — CHANGES_REQUESTED** [review]
I still don't see any reason why we would need to change the way we import/export the tunnel launcher package.

**Comment by hieuxlu** [comment]
> I still don't see any reason why we would need to change the way we import/export the tunnel launcher package.

I don't really understand. The migration of this package to Typescript caused a breaking change because Typescript by - default transpile  `import from` as `import default`. This does not work with packages using `export =` so we need to use correct `import require` for this package.

**Comment by christian-bromann** [comment]
> This does not work with packages using `export =`

The `testingbot-tunnel-launcher` does not use this kind of export. If you look into the build code of that package you see at the end of the file:

```js
module.exports = downloadAndRun;
module.exports.kill = killTunnel;
```

This means we can just import it like any other package via

```js
import TunnelLauncher from 'testingbot-tunnel-launcher'
```

Who is using `export =`?

**Comment by hieuxlu** [comment]
Isn't that the same? The exported result is a function and not es6 module.

**Comment by christian-bromann** [comment]
> Isn't that the same?

No, `export = ...` is different from `module.exports`. A lot of packages this project depends on exports their modules this way and we also don't need to treat them differently. Also if you note, none of the proposed changes in #6105 was related to import/export issues of the tunnel package.

**Comment by hieuxlu** [comment]
> > Isn't that the same?
> 
> No, `export = ...` is different from `module.exports`. A lot of packages this project depends on exports their modules this way and we also don't need to treat them differently. Also if you note, none of the proposed changes in #6105 was related to import/export issues of the tunnel package.

I will have a look. The changes proposed there only happen because they can not use the tunnel library at all, so they had to generate tunnel id randomly, and remove the tunnel code entirely

**Comment by hieuxlu** [comment]
@christian-bromann You're probably right, I can't reproduce the original issue any more. I don't know what happened, but I'll reopen the original PR

**Comment by hieuxlu** [comment]
I still don't understand why I can't reproduce the issue. It's very clear to me from the docs that `import require` is the only way that'll work with `module.exports =` or `export =`

https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require
```
The export = syntax specifies a single object that is exported from the module. This can be a class, interface, namespace, function, or enum.

When exporting a module using export =, TypeScript-specific import module = require("module") must be used to import the module.
```

**Comment by christian-bromann** [comment]
@hieuxlu can you please tell me where you see `export =` being used??? I don't see this at all, neither in the compiled code of the `testingbot-tunnel-launcher` nor in any of our sub packages.

**Comment by hieuxlu** [comment]
> @hieuxlu can you please tell me where you see `export =` being used??? I don't see this at all, neither in the compiled code of the `testingbot-tunnel-launcher` nor in any of our sub packages.

It's a Typescript syntax that will be compiled into `module.exports =`

**Comment by hieuxlu** [comment]
Hey, sorry the the heated discussion, but I have no idea why I can't reproduce this any more. In addition, the original issue I think was a misconfiguration with wdio v6 because of root-level config breaking changes, so there were no bug, just a wrong type definition. So once testingbot confirmed, I will correct the import require. Thanks.

**Comment by christian-bromann** [comment]
Given we merged #6105 I think this got obsolete now.

File: packages/wdio-testingbot-service/tests/__mocks__/testingbot-tunnel-launcher.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-export default (opts: any, cb: any) => {
+export = (opts: any, cb: any) => {
     cb(null, {
```
Comment (hieuxlu) [inline packages/wdio-testingbot-service/tests/__mocks__/testingbot-tunnel-launcher.ts:2]: @christian-bromann This was what made me confused. The mock for Jest uses `export default`, so I was confused.

Comment (hieuxlu) [reply packages/wdio-testingbot-service/tests/__mocks__/testingbot-tunnel-launcher.ts:2]: So if it's OK, I'd propose to fix the unit test mocking for consistency in this PR. I'll revert unrelated codes.

Comment (christian-bromann) [reply packages/wdio-testingbot-service/tests/__mocks__/testingbot-tunnel-launcher.ts:2]: Why does the mock need to be changed? `export default` is correct here as it mimics the way how testingbot exports its service: `module.exports = require('./build/tunnel-launcher')`

**Comment by hieuxlu** [comment]
Typescript is not treating them the same. But I guess it's fine to leave it
as is. Two wrong is one right, the definition file use export default too
so nothing breaks. I'll leave it here.

**Pull request closed without merging**